media-libs/libdiscid/patches/libdiscid-0.6.2.patchset

```diff
@@ -0,0 +1,180 @@
++endif
+ if OS_DARWIN
+ libdiscid_la_LDFLAGS += -framework CoreFoundation -framework IOKit
+ libdiscid_la_SOURCES += src/toc.c src/unix.c src/disc_darwin.c
+diff --git a/configure.ac b/configure.ac
+index 0d21232..accef71 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -45,7 +45,7 @@ AC_DEFINE(libdiscid_EXPORTS, [], "the header is used to create the library")
+ 
+ 
+ case "${host_cpu}-${host_os}" in
+-  #*-beos*)    os=beos;;
++  *-haiku*)    os=haiku;;
+   *-mingw*)   os=win32;;
+   *-cygwin*)  os=win32;;
+   *-freebsd*) os=freebsd ;;
+@@ -63,7 +63,7 @@ case "${host_cpu}-${host_os}" in
+ esac
+ 
+ AC_MSG_NOTICE([Using disc read implementation for: $os])
+-#AM_CONDITIONAL([OS_BEOS], [test x$os = xbeos])
++AM_CONDITIONAL([OS_HAIKU], [test x$os = xhaiku])
+ AM_CONDITIONAL([OS_DARWIN], [test x$os = xdarwin])
+ AM_CONDITIONAL([OS_FREEBSD], [test x$os = xfreebsd])
+ AM_CONDITIONAL([OS_GENERIC], [test x$os = xgeneric])
+diff --git a/src/disc_haiku.c b/src/disc_haiku.c
+new file mode 100644
+index 0000000..6c07574
+--- /dev/null
++++ b/src/disc_haiku.c
+@@ -0,0 +1,125 @@
++/* --------------------------------------------------------------------------
++
++   MusicBrainz -- The Internet music metadatabase
++
++   Copyright (C) 2009 Shunsuke Kuroda
++   Copyright (C) 2006 Matthias Friedrich
++   Copyright (C) 2000 Robert Kaye
++   Copyright (C) 1999 Marc E E van Woerkom
++   
++   This library is free software; you can redistribute it and/or
++   modify it under the terms of the GNU Lesser General Public
++   License as published by the Free Software Foundation; either
++   version 2.1 of the License, or (at your option) any later version.
++   
++   This library is distributed in the hope that it will be useful,
++   but WITHOUT ANY WARRANTY; without even the implied warranty of
++   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
++   Lesser General Public License for more details.
++   
++   You should have received a copy of the GNU Lesser General Public
++   License along with this library; if not, write to the Free Software
++   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
++   02110-1301  USA
++
++--------------------------------------------------------------------------- */
++#include <stdlib.h>
++#include <stdio.h>
++#include <string.h>
++#include <fcntl.h>
++#include <assert.h>
++#include <unistd.h>
++#include <scsi.h>
++#include <sys/types.h>
++#include <sys/stat.h>
++#include <sys/ioctl.h>
++
++
++#include "discid/discid.h"
++#include "discid/discid_private.h"
++#include "unix.h"
++
++#define NUM_CANDIDATES 4
++
++static char *device_candidates[NUM_CANDIDATES] = {
++	"/dev/disk/atapi/0/master/raw",
++	"/dev/disk/atapi/0/slave/raw",
++	"/dev/disk/atapi/1/master/raw",
++	"/dev/disk/atapi/1/slave/raw"
++};
++
++int mb_disc_unix_read_toc_header(int fd, mb_disc_toc *disk_toc) {
++	scsi_toc toc;
++	int ret = ioctl(fd, B_SCSI_GET_TOC, &toc);
++
++	if ( ret < 0 )
```

Comment:
According to the POSIX [ioctl()][0], only `-1` is considered an error code, values other than that indicates success

```diff
-+	if ( ret < 0 )
++	if ( ret == -1 )
```

[0]: http://pubs.opengroup.org/onlinepubs/009695399/functions/ioctl.html#tag_03_282_04